lib/Profile.js

```diff
@@ -5,7 +5,7 @@ function Profile(data, raw) {
 
   if (data.identities) {
     this.provider = data.identities[0].provider;
-  } else {
+  } else if (typeof this.id === 'string' && this.id.indexOf('|') > -1 ) {
```

Comment:
Note this will only work if the user id (sub) prepends the provider name. There are scenarios, like custom DB scripts, that choose the format of the user id and may not include a `|` separator or the provider name at all. As it is now on this PR, the "else" scenario will be not setting a provider key. I guess that is fine over exploding on runtime 😆